src/engine.rs

```diff
@@ -28,13 +28,15 @@ use crate::node::PbftNode;
 use crate::state::{PbftMode, PbftState};
 use crate::storage::get_storage;
 use crate::timing;
+use crate::PbftCliArgs;
 
-#[derive(Default)]
-pub struct PbftEngine {}
+pub struct PbftEngine {
+    cli_args: PbftCliArgs,
+}
 
 impl PbftEngine {
-    pub fn new() -> Self {
-        PbftEngine {}
+    pub fn new(cli_args: PbftCliArgs) -> Self {
```

Comment:
I think you should pass in the pbft config here and keep the args used only in main.rs. Otherwise, it would be awkward to use PbftEngine as library code.

Reply:
Fixed